packages/valory/skills/twitter_scoring_abci/behaviours.py

```diff
@@ -738,19 +738,31 @@ def update_ceramic_db(self) -> Dict:
 
         return ceramic_db.data
 
+    def get_registration(self, text: str) -> Optional[str]:
+        """Check if the tweet is a registration and return the wallet address"""
 
-def get_registration(text: str) -> Optional[str]:
-    """Check if the tweet is a registration and return the wallet address"""
+        wallet_address = None
 
-    wallet_address = None
+        address_match = re.search(ADDRESS_REGEX, text)
+        tagline_match = re.search(TAGLINE, text, re.IGNORECASE)
 
-    address_match = re.search(ADDRESS_REGEX, text)
-    tagline_match = re.search(TAGLINE, text, re.IGNORECASE)
+        if address_match and tagline_match:
+            wallet_address = Web3.to_checksum_address(address_match.group())
 
-    if address_match and tagline_match:
-        wallet_address = Web3.to_checksum_address(address_match.group())
+            address_to_twitter_handles = {
+                user["wallet_address"]: user["twitter_handle"]
+                for user in self.synchronized_data.ceramic_db["users"]
+                if user["wallet_address"]
+            }
+
+            # Ignore registration if both address and Twitter handle already exist
+            if (
+                wallet_address in address_to_twitter_handles
+                and address_to_twitter_handles[wallet_address]
+            ):
+                return None
```

Comment:
If the `ceramic_db` contains two users with the same wallet address and different handles, then this logic might fail to register one of the handles. 
Can this ever happen, i.e., the db containing two users with the same address and different handles?
If yes, is it intentional not to register one of the handles?

Reply:
We do not allow the same address being registered with two users. In fact, we have some logic to merge users that share the same address.

Reply:
We don't do it for every tweet , but for every registration tweet, which are not very common (users only post those tweets when they register to the service). Anyway, I also thought about keeping it in memory, but since this adds to the pile of data that the service needs to keep updated and in sync, and given that we are going to revamp the registration system with a proper signature-based one, I decided to go this way.